nodeup/pkg/model/kube_apiserver.go

```diff
@@ -316,19 +316,29 @@ func (b *KubeAPIServerBuilder) buildPod() (*v1.Pod, error) {
 		}
 	}
 
+	var mainEtcdCluster, eventsEtcdCluster string
+	if b.IsMaster {
+		mainEtcdCluster = "https://127.0.0.1:4001"
+		eventsEtcdCluster = "https://127.0.0.1:4002"
+	} else {
+		host := b.Cluster.ObjectMeta.Name
+		mainEtcdCluster = "https://main.etcd." + host + ":4001"
+		eventsEtcdCluster = "https://events.etcd." + host + ":4002"
```

Comment:
I'm wondering if these should use `b.Cluster.Spec.MasterPublicName` or `b.Cluster.Spec.MasterInternalName` instead of `b.Cluster.ObjectMeta.Name`

Reply:
so `main.etcd.api.test.example.com`?
If they do not directly relate to api nodes (and in the future, may not even run api servers), does that make sense?

Reply:
hm you're right, having `api` in there doesn't make sense. Maybe down the road we expose a third control plane name field in the clusterspec. If users are wanting to override the default master name fields, they'll likely want to override these etcd names as well.

Reply:
I think there were someone wanting this field also for the newly-introduced kops-controller DNS. I can try to use the same approach as we did there (which was replace "api" with "kops-controller", I believe).

The introduction of the suffix field can be done in a follow-up.

Reply:
Seems to have remembered incorrectly. kops-controller does `dns.alpha.kubernetes.io/internal: kops-controller.internal.{{ ClusterName }}`

So maybe stick with current, but add support for configurable prefix later?